### PR TITLE
[Pokemon] Fixes and updates

### PR DESCRIPTION
**Pull request by Alchav**
## What is this fixing or adding?

Fixes logic bugs reported by j_im
Fixes to account for 21 byte seed names when generating on web host
Filler item update
Number of badges needed to enter Viridian Gym displayed in game when attempting to enter

## How was this tested?

Generated games and played to ensure text works and no errors with logic changes.
Generated games locally and on webhost to ensure connection would not be rejected due to mismatched seed name

### CODE REVIEW

File: worlds/pokemon_rb/rom.py
Comment (Berserker66) [inline worlds/pokemon_rb/rom.py:548]: it's 20 here, but 21 in the lua?

Comment (Alchav) [reply worlds/pokemon_rb/rom.py:548]: I meant to remove this part. Now I don't know why it didn't fail

Comment (Alchav) [reply worlds/pokemon_rb/rom.py:548]: Oh, this is for the seed number displayed on the title screen in game. There isn't room for a 21st character

**Comment by Alchav** [comment]
Until a client is available with these changes, no one will be able to play Pokemon R/B on any game generated on the webhost. However once this PR is merged, no one will be able to play any Pokemon R/B games at all until a client with these changes is available. I know people have been jumping into beta tests for other games to play Pokemon since it's on source now. I'm hoping RCs will be soon and this could be merged soon before the RC

**Comment by Berserker66** [comment]
So I should hold off on merging this until right before the RC?

**Comment by Alchav** [comment]
I suppose it depends on how long that would be from now. If it's going to be a while, I could just build source and distribute it for pokemon players after this is merged

**Comment by Alchav** [comment]
The bug present in the earlier beta test appears to be fixed. I believe in fixing the stray commit I overwrote the base patches with earlier versions which did not match with the rom address pointers.

**Comment by Alchav** [comment]
Since I have released a Pokemon Client off this branch as it is, I'm going to recommend merging it now so that my released client works on other current-source beta test games

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**
